sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala

```diff
@@ -83,10 +83,13 @@ private[sql] class JSONRelation(
     val job = new Job(sqlContext.sparkContext.hadoopConfiguration)
     val conf = SparkHadoopUtil.get.getConfigurationFromJobContext(job)
 
-    val paths = inputPaths.map(_.getPath)
+    val rddInputPaths = inputPaths.map(_.getPath)
 
-    if (paths.nonEmpty) {
-      FileInputFormat.setInputPaths(job, paths: _*)
+    if (rddInputPaths.nonEmpty) {
+      FileInputFormat.setInputPaths(job, rddInputPaths: _*)
+    } else {
+      throw new RuntimeException("Input paths do not exist or are empty directories, "
```

Comment:
Use `require` to write this test. In general `RuntimeException` is not something you should throw directly.